cognite/neat/graph/extractors/_iodd.py

```diff
@@ -0,0 +1,149 @@
+import xml.etree.ElementTree as ET
+from pathlib import Path
+from typing import ClassVar
+from xml.etree.ElementTree import Element
+
+from rdflib import RDF, Literal, Namespace, URIRef
+
+from cognite.neat.constants import DEFAULT_NAMESPACE
+from cognite.neat.graph.extractors._base import BaseExtractor
+from cognite.neat.graph.models import Triple
+from cognite.neat.utils.rdf_ import as_neat_compliant_uri
+from cognite.neat.utils.text import to_camel
+from cognite.neat.utils.xml_ import get_children_from_tag
+
+IODD = Namespace("http://www.io-link.com/IODD/2010/10/")
+XSI = Namespace("http://www.w3.org/2001/XMLSchema-instance/")
+
+
+class IODDExtractor(BaseExtractor):
+    """
+    IODD-XML extractor of RDF triples
+
+    Each IODD sheet describes an IODD device. This extractor extracts rdf triples that describes the device, and the
+    sensors connected to the device.
+    This data is described under the elements "DeviceIdentity" and "ProcessDataCollection".
+    In addition, triples extacted from "DeviceIdentity" and
+    "ProcessDataCollection" may reference "Text" elements which are found under "ExternalTextCollection". Edges to
+    these Text element nodes are also extracted.
+
+    Args:
+        root: XML root element of IODD XML file.
+        namespace: Optional custom namespace to use for extracted triples that define data
+                    model instances. Defaults to DEFAULT_NAMESPACE.
+    """
+
+    device_elements_with_text_nodes: ClassVar[list[str]] = ["VendorText", "VendorUrl", "DeviceName", "DeviceFamily"]
+
+    def __init__(self, root: Element, namespace: Namespace | None = None):
+        self.root = root
+        self.namespace = namespace or DEFAULT_NAMESPACE
+
+    @classmethod
+    def from_file(cls, filepath: str | Path, namespace: Namespace | None = None):
+        return cls(ET.parse(filepath).getroot(), namespace)
+
+    @classmethod
+    def _from_root2triples(cls, root: Element, namespace: Namespace) -> list[Triple]:
```

Comment:
As opposed to the dexpi extractor which parses one element at a time and yields its triples, I've implemented this extractor to extract all triples starting from the root and returning the entire list of triples. This is mainly because I'm only interested in three sections of the XML sheet as seen below. Let me know however, if you think this should be revised @nikokaoja